### PR TITLE
Chore/sc 519/upgrade swr to 1 0

### PR DESCRIPTION
**Pull request by adamgoth**
### Link to Clubhouse story

https://app.shortcut.com/dux-makerdao/story/519/upgrade-swr-to-1-0

### What does this PR do?

Upgrades to SWR 1.0

Reduces default polling interval

Cuts down on unnecessary calls

Adds mutations (revalidation) where necessary

### Steps for testing:

Check all the pages of the app. Send tests transactions. Look for anything funky.

### Any additional helpful information?:

Reducing infura calls will be handled in a separate PR

### CODE REVIEW

File: modules/executive/hooks/useSpellData.ts [hunk 1]
```diff
@@ -6,16 +6,18 @@ type SpellDataResponse = {
...
+  const { data, error, mutate } = useSWR<SpellData>(
     `/api/executive/analyze-spell/${proposalAddress}?network=${getNetwork()}`
   );
 
   return {
     data,
     loading: !error && !data,
-    error
+    error,
+    mutate
```
Comment (rafinskipg) [inline modules/executive/hooks/useSpellData.ts:21]: nice

**Review by rafinskipg — CHANGES_REQUESTED** [review]
So it seems on the delegates page, if you freshload, it does not load the MKR delegated, but if you switch focus ( change to other app or tab) it does refresh the balances

**Comment by adamgoth** [comment]
> So it seems on the delegates page, if you freshload, it does not load the MKR delegated, but if you switch focus ( change to other app or tab) it does refresh the balances

Nice catch. Should be fixed now  🔨

File: lib/hooks/useDelegateAddressMap.ts [hunk 2]
```diff
@@ -8,7 +8,9 @@ type DelegateAddressMapResponse = {
 };
 
 export const useDelegateAddressMap = (): DelegateAddressMapResponse => {
-  const { data: delegatesApiResponse, error } = useSWR(`/api/delegates?network=${getNetwork()}`);
+  const { data: delegatesApiResponse, error } = useSWR(`/api/delegates?network=${getNetwork()}`, null, {
+    refreshInterval: 0
```
[the remark this answers is not in the document]
Comment (adamgoth) [reply lib/hooks/useDelegateAddressMap.ts:12]: Exactly

File: modules/delegates/components/modals/DelegateModal.tsx [hunk 3]
```diff
@@ -68,6 +76,8 @@ export const DelegateModal = ({ isOpen, onDismiss, delegate }: Props): JSX.Eleme
     const lockTxCreator = () => maker.service('voteDelegate').lock(voteDelegateAddress, mkrToDeposit);
     const txId = await trackTransaction(lockTxCreator, 'Depositing MKR', {
       mined: txId => {
+        mutateTotalStaked();
+        mutateMkrStaked();
```
[the remark this answers is not in the document]
Comment (adamgoth) [reply modules/delegates/components/modals/DelegateModal.tsx:80]: Not on modal close. When we broadcast the tx, we track it through dai.js's transaction manager (through `trackTranscation`). That tracking function provides hooks for when the tx status changes. So as soon as we detect that the tx was mined, that's when those functions trigger.

**Review by rafinskipg — APPROVED** [review]

**Pull request merged**